interview-questions.md

```diff
@@ -8,44 +8,60 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is a hash in Ruby?
 
-Your answer:
-
-Researched answer:
+Your answer: A ruby hash is a unique instantiated object that looks and operates similarly to a standard object you might see in JS or an array. The difference is in how the hash indexes its key-value pairs. This indexing, as apposed to using integers, uses arbitrary values that are made difficult to access/ traverse as a built in security feature. Hashes have a great amount of applications in storing data in this way and are the standard for holding important data in many programs. The main reason this feature improves security is in the specificity in which you call keys. Using the wrong key will simply return nil.
+
+Researched answer: After some greater research, hashes can have interesting implementations when used as empty method parameters. Using ruby hashes built in method .empty? you can create dynamic functionality within your methods. The example below shows the mutable nature to which you can use hashes:
+```ruby
+def user_info(name, optional_info = {})
+    if optional_info.empty?
+        p "Users Name: #{name}"
+    else
+        p "Users Name: #{name}, Users Address: #{optional_info[:address]}, User Phone: #{optional_info[:phone_num]}"
+    end
+end
+
+user_info 'Joe'
+user_info('Joe', {address: '123 Joe St.', phone_num: 1233333333})
+```
 
 2. What is a gem?
 
-Your answer:
+Your answer: A gem in ruby is a package of code that can be applied to a project to access a special "toolkit" of libraries and programs. In general gems are libraries that can be used to expedite the processes that could potentially be time consuming for a developer to rewrite. In general its ideal to not have to rewrite code thats already been written, and in the open source culture of ruby, these gems are a corner stone to using ruby. 
+
+Researched answer: After greater research its important to note that anyone can create a collection of scripts to serve as a gem and upload it for open source use. The file extension .gem is used to detail a built gem which can be done using the terminal command:
+
```

Comment:
"anyone"...makes me want to analyze the gems a bit more. What are precautions that you take when loading gems?